includes/payment-methods/Configs/Definitions/AlipayDefinition.php

```diff
@@ -0,0 +1,308 @@
+<?php
+/**
+ * Alipay Payment Method Definition
+ *
+ * @package WCPay\PaymentMethods\Configs\Definitions
+ */
+
+namespace WCPay\PaymentMethods\Configs\Definitions;
+
+use WCPay\PaymentMethods\Configs\Interfaces\PaymentMethodDefinitionInterface;
+use WCPay\PaymentMethods\Configs\Constants\PaymentMethodCapability;
+use WCPay\Constants\Country_Code;
+use WCPay\Constants\Currency_Code;
+use WCPay\PaymentMethods\Configs\Utils\PaymentMethodUtils;
+
+/**
+ * Class implementing the Alipay payment method definition.
+ */
+class AlipayDefinition implements PaymentMethodDefinitionInterface {
+
+	/**
+	 * Get the internal ID for the payment method
+	 *
+	 * @return string
+	 */
+	public static function get_id(): string {
+		return 'alipay';
+	}
+
+	/**
+	 * Get the keywords for the payment method
+	 *
+	 * @return string[]
+	 */
+	public static function get_keywords(): array {
+		return [ 'alipay' ];
+	}
+
+	/**
+	 * Get the Stripe payment method ID
+	 *
+	 * @return string
+	 */
+	public static function get_stripe_id(): string {
+		return PaymentMethodUtils::get_stripe_id( self::get_id() );
+	}
+
+	/**
+	 * Get the customer-facing title of the payment method
+	 *
+	 * @param string|null $account_country Optional. The merchant's account country.
+	 *
+	 * @return string
+	 */
+	public static function get_title( ?string $account_country = null ): string {
+		return __( 'Alipay', 'woocommerce-payments' );
+	}
+
+	/**
+	 * Get the customer-facing description of the payment method
+	 *
+	 * @return string
+	 */
+	public static function get_description(): string {
+		return __( 'Alipay is a popular wallet in China, operated by Ant Financial Services Group, a financial services provider affiliated with Alibaba.', 'woocommerce-payments' );
+	}
+
+	/**
+	 * Is the payment method a BNPL (Buy Now Pay Later) payment method?
+	 *
+	 * @return boolean
+	 */
+	public static function is_bnpl(): bool {
+		return PaymentMethodUtils::is_bnpl( self::get_capabilities() );
+	}
+
+	/**
+	 * Is the payment method a reusable payment method?
+	 *
+	 * @return boolean
+	 */
+	public static function is_reusable(): bool {
+		return PaymentMethodUtils::is_reusable( self::get_capabilities() );
+	}
+
+	/**
+	 * Does the payment method accept only domestic payments?
+	 *
+	 * @return boolean
+	 */
+	public static function accepts_only_domestic_payments(): bool {
+		return PaymentMethodUtils::accepts_only_domestic_payments( self::get_capabilities() );
+	}
+
+	/**
+	 * Does the payment method allow manual capture?
+	 *
+	 * @return boolean
+	 */
+	public static function allows_manual_capture(): bool {
+		return PaymentMethodUtils::allows_manual_capture( self::get_capabilities() );
+	}
+
+	/**
+	 * Get the list of supported currencies
+	 *
+	 * @return string[] Array of currency codes
+	 */
+	public static function get_supported_currencies(): array {
```

Comment:
@frosso  I'm not sure if this implementation is accurate, from what I understand is that CNY is available in all countries along each country's own local currency https://docs.stripe.com/payments/alipay#supported-currencies

We do something similar for WeChat Pay https://github.com/Automattic/woocommerce-payments/blob/1957d7a308f5ae32a272b907dc4ac6ab132296bd/includes/payment-methods/class-wechatpay-payment-method.php#L107-L156

Reply:
@danielmx-dev from my understanding, we support a subset of the currencies that Stripe will support for each country.
I was looking at the 2025 pricing sheet and at the Fees tool in MC. I see only one currency for each of the countries, and I don't see `CNY` added for each country.
For this reason, I didn't add `CNY` to each account country.

Am I reading that right?